### PR TITLE
udev: reduce number of times checking event dependency

### PR DESCRIPTION
**Pull request by yuwata**
Most commits are trivial cleanup.

### CODE REVIEW

File: src/udev/udev-ctrl.c [hunk 1]
```diff
@@ -23,14 +23,14 @@
 /* wire protocol magic must match */
 #define UDEV_CTRL_MAGIC                                0xdead1dea
```
Comment (keszybz) [inline src/udev/udev-ctrl.c:24]: That's ominous. ;)

File: src/udev/udevd.c [hunk 2]
```diff
@@ -992,7 +992,7 @@ static int on_uevent(sd_device_monitor *monitor, sd_device *dev, void *userdata)
         }
 
         /* we have fresh events, try to schedule them */
-        event_queue_start(manager);
+        (void) event_queue_start(manager);
```
Comment (keszybz) [inline src/udev/udevd.c]: Please drop the (void) casts, as discussed in https://github.com/systemd/systemd/pull/20138#discussion_r669340803.

File: src/basic/list.h [hunk 3]
```diff
@@ -145,9 +145,15 @@
...
         for ((i) = (p)->name##_prev; (i); (i) = (i)->name##_prev)
 
+#define LIST_FOREACH_BEFORE_FROM(name,i,p)                              \
+        for ((i) = (p); (i); (i) = (i)->name##_prev)
+
 #define LIST_FOREACH_AFTER(name,i,p)                                    \
         for ((i) = (p)->name##_next; (i); (i) = (i)->name##_next)
 
+#define LIST_FOREACH_AFTER_FROM(name,i,p)                               \
+        for ((i) = (p); (i); (i) = (i)->name##_next)
```
Comment (keszybz) [inline src/basic/list.h]: I find those names very cryptic… How is `LIST_FOREACH_AFTER_FROM` different than `LIST_FOREACH` ?

Wouldn't it be better to keep just a small set of iteration macros, e.g. LIST_FOREACH and LIST_FOREACH_BACKWARDS, and replace LIST_FOREACH_BEFORE() with LIST_FOREACH_BACKWARDS(foo.name_prev), and LIST_FOREACH_BEFORE_FROM() with LIST_FOREACH_BACKWARDS(foo) (and similarly for the forward versions) ?

**Review by keszybz — CHANGES_REQUESTED** [review]

File: src/udev/udevd.c (hunk 2 above)
Comment (yuwata) [reply src/udev/udevd.c]: OK, at least tentatively.

**Comment by yuwata** [comment]
@keszybz Thank you for your review. The above points are addressed. PTAL.

**Review by keszybz — COMMENTED** [review]
I don't fully grok the event dependency logic, but the code looks OK.

**Comment by keszybz** [comment]
@poettering, @bluca maybe you could review this too, esp. the last two commits.

File: src/udev/udevd.c [hunk 4]
```diff
@@ -770,15 +770,16 @@ static int event_run(Event *event) {
...
 
-/* lookup event for identical, parent, child device */
-static int is_device_busy(Manager *manager, Event *event) {
+static int event_is_blocked(Event *event) {
         const char *subsystem, *devpath, *devpath_old = NULL;
         dev_t devnum = makedev(0, 0);
         Event *loop_event;
         size_t devpath_len;
         int r, ifindex = 0;
         bool is_block;
```
Comment (bluca) [inline src/udev/udevd.c:780]: `assert(event)`?

Comment (bluca) [reply src/udev/udevd.c:780]: nvm, added later

**Review by bluca — APPROVED** [review]
> @poettering, @bluca maybe you could review this too, esp. the last two commits.

from what little I understand about udev this looks fine to me - but it's very, very little :-)

**Pull request merged**